src/lib/select/select.ts

```diff
@@ -1119,6 +1119,11 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
     const maxOptionsDisplayed = Math.floor(SELECT_PANEL_MAX_HEIGHT / itemHeight);
     let optionOffsetFromPanelTop: number;
 
+    // Disable offset for smaller lists by returning 0 as value to offset
+    if (this._getItemCount() <= maxOptionsDisplayed) {
```

Comment:
This goes against the [Material design spec](https://storage.googleapis.com/material-design/publish/material_v_12/assets/0B6Okdz75tqQsbXpoZkNIcXgxc3M/components-dialogs-simplemenus3.png) and I'm expecting it to break a lot of tests once we get to syncing it in. I think it's fine to have a functionality to disable it, but it should be an input that the consumer can opt-out of.

Reply:
Thanks for the feedback. I've updated the code to use a custom property to disable the offset only if the consumer chooses to do so by specifying it as a custom property. If nothing is specified, then the offset is enabled by default.